assets/apps/customizer-controls/src/global-custom-colors/PaletteColors.js

```diff
@@ -0,0 +1,208 @@
+	const newColorSlot = () => {
+		if (slugs.length === 0) {
+			return 1;
+		}
+
+		const customSlugs = slugs.filter((slug) => {
+			return slug.substring(0, 7) === 'custom-';
+		});
+
+		const customSlugNumbers = customSlugs.map((slug) => {
+			return parseInt(slug.substring(7));
+		});
+
+		customSlugNumbers.sort((a, b) => {
+			return a - b;
+		});
+
+		return customSlugNumbers.at(-1) + 1;
+	};
+
+	const updateColorInPalette = (colorSlug, label, val) => {
+		const nextValues = { ...values };
+		if (nextValues[colorSlug].val === val) {
+			return false;
+		}
+		nextValues[colorSlug].label = label;
+		nextValues[colorSlug].val = val;
+		save(nextValues);
+	};
+
+	const deleteColor = () => {
+		if (!willDelete) {
+			return;
+		}
+
+		const nextValues = { ...values };
+		delete nextValues[willDelete];
+		save(nextValues);
+		closeDeleteModal();
+	};
+
+	const editLabel = (slug, newLabel) => {
+		const nextValues = { ...values };
+		if (nextValues[slug].label === newLabel) {
+			setWillEdit('');
+			return false;
+		}
+		nextValues[slug].label = newLabel;
+		save(nextValues);
+		setWillEdit('');
+	};
+
+	const editLabelCancel = () => setWillEdit('');
+
+	return (
+		<>
+			<Accordion label={__('Custom Colors', 'neve')}>
+				<div className="color-array-wrap">
+					{slugs.map((slug) => {
+						if (slug === 'flag') {
+							return null;
+						}
+
+						const isEditing = willEdit === slug;
+
+						return (
+							<ColorControl
+								disableGlobal
+								key={slug}
+								label={!isEditing ? values[slug].label : null}
+								selectedColor={values[slug].val}
+								onChange={debounce((value) => {
+									updateColorInPalette(
+										slug,
+										values[slug].label,
+										value
+									);
+								}, 100)}
+							>
+								{isEditing && (
+									<EditColorLabel
+										currentLabel={values[slug].label}
+										slug={slug}
+										save={editLabel}
+										cancel={editLabelCancel}
+									/>
+								)}
+								{!isEditing && (
+									<div className="action-icon-wrapper">
+										<Icon
+											onClick={() => {
+												setWillEdit(slug);
+											}}
+											icon="edit"
+										/>
+									</div>
+								)}
+								<div className="action-icon-wrapper">
+									<Icon
+										onClick={() => {
+											setWillDelete(slug);
+											openDeleteModal();
+										}}
+										icon="trash"
+									/>
+								</div>
+							</ColorControl>
+						);
+					})}
+					<Button
+						className="new-custom-color"
+						disabled={isHitLimit}
+						onClick={() => {
+							addNewColor();
+						}}
+						variant="secondary"
+					>
+						{__('Add Custom Color', 'neve')}
+						<Icon icon={plus} />
+					</Button>
+					{isHitLimit && (
+						<div className="cc-limit-notice">
+							<Icon icon={warning} />
+							<p>
+								{sprintf(
+									// translators: %d: Allowed maximum custom color
+									__(
+										'Maximum %d custom colors can be added.',
```

Comment:
```suggestion
										'Maximum %d %s can be added.',
```
We can use this for a more general context.

Reply:
Thanks!